### PR TITLE
Build srcs that are other build targets

### PR DESCRIPTION
**Pull request by Tatskaari**

### CODE REVIEW

**Comment by sfirmery** [comment]
puku seems to remove `srcs` that are not existing files. Does this PR solve this problem too?

**Comment by Tatskaari** [comment]
I'm not sure if it does actually. I'll add a test for that next week.

**Comment by Tatskaari** [comment]
Okay, can confirm this fixes that issue :)

File: generate/generate.go [hunk 1]
```diff
@@ -364,26 +360,40 @@ func newModDownloadRule(mod, version string, licences []string) (*build.CallExpr
 	return rule.Call, rule.Name()
 }
 
-func (u *Update) allSources(r *rule) ([]string, error) {
-	args := r.parseGlob()
-	if args == nil {
-		return r.AttrStrings("srcs"), nil
+func (u *Update) allSources(conf *config.Config, r *rule, sourceMap map[string]*GoFile) ([]string, map[string]*GoFile, error) {
```
Comment (goddenrich) [inline generate/generate.go:363]: this function could do with a docstring to explain what the return values are

Comment (Tatskaari) [reply generate/generate.go]: Okay, done :)

**Review by samwestmoreland — APPROVED** [review]
Looks good to me!

**Comment by sfirmery** [comment]
@Tatskaari 

I've tried this PR in my environment and found that:
- using variables in `srcs` seems to not work, like using `codegen = text_file()`
- output files that are in a `provides: {"go": ":label"}` of a label in `srcs` seems not used

**Comment by Tatskaari** [comment]
> using variables in srcs seems to not work, like using codegen = text_file()

Yeah this is best effort. I want to write a proper evaluator but I think i'm going to need to become persuasive to get that into the roadmap haha

> output files that are in a provides: {"go": ":label"} of a label in srcs seems not used

Ooff yeah that's a good point. I will have to think about this. Thanks for the feedback though.

**Comment by sfirmery** [comment]
Yes, in my use case, I use a `proto_library()` to only generate go files then use them and other non generated go files in a `go_library()`.

**Comment by Tatskaari** [comment]
Ah, fair enough. I've raised an issue here:
#62 
however I think this might be better off living in https://github.com/please-build/buildtools

**Pull request merged**